components/adminTable.js

```diff
@@ -44,12 +47,42 @@ export default function AdminTable(props) {
   const { getTableProps, getTableBodyProps, headerGroups, rows, prepareRow } =
     useTable({ columns, data });
 
+  const paginatedRows = React.useMemo(
+    () =>
+      rows.slice(pageIndex * entriesPerPage, (pageIndex + 1) * entriesPerPage),
+    [rows, pageIndex, entriesPerPage]
+  );
+
+  const startEntry = pageIndex * entriesPerPage + 1;
+  const endEntry = Math.min((pageIndex + 1) * entriesPerPage, rows.length);
+  const totalEntries = rows.length;
+
+  // Dynamically generate sensible options for rows per page
+  const baseOptions = [10, 20, 50, 100];
+  let entriesPerPageOptions = baseOptions.filter(opt => opt < totalEntries);
+  if (totalEntries < 100 && !entriesPerPageOptions.includes(totalEntries)) {
+    entriesPerPageOptions.push(totalEntries);
+  }
+  // If totalEntries is less than the smallest option, just show totalEntries
+  if (totalEntries < 10) {
+    entriesPerPageOptions = [totalEntries];
+  }
+
+  const canPreviousPage = pageIndex > 0;
+  const canNextPage = endEntry < totalEntries;
+
+  const getPaginationButtonStyle = disabled => ({
+    marginLeft: '10px',
+    marginRight: '10px',
+    color: disabled ? '#d1d1d1' : '#757575',
+    fontSize: '1.5em'
+  });
```

Comment:
When declaring a css style rule that you wish to reuse, a better practice is to create a separate css file for the module.